src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -48,24 +53,115 @@
 @Log4j2
 public class HybridQueryPhaseSearcher extends QueryPhaseSearcherWrapper {
 
+    final static int MAX_NESTED_SUBQUERY_LIMIT = 50;
```

Comment:
rather than hardcoding it here, can we read index settings?

Reply:
ack